mlrun/runtimes/function.py

```diff
@@ -209,14 +208,12 @@ def __init__(
         self.no_cache = no_cache
         self.readiness_timeout = readiness_timeout
 
-        # TODO: we would prefer to default to 0, but invoking a scaled to zero function requires to either add the
-        #  x-nuclio-target header or to create the function with http trigger and invoke the function through it - so
-        #  we need to do one of the two
         self.min_replicas = min_replicas or 1
-        self.max_replicas = max_replicas or default_max_replicas
+        self.max_replicas = max_replicas or 4
```

Comment:
lets make it configurable, for nuclio default min and max replicas

Reply:
I am not sure it is the right time to do it, mainly because it overload the configuration and nothing will override it

Reply:
I actually think we might want to even disable that auto enrichment as it block users to create function with scale to zero as if `min_replicas` will be `0` it will get overridden with `1`

Reply:
yup, dif pr